tests/L0/run_transformer/test_weight_grad.py

```diff
@@ -0,0 +1,135 @@
+import torch
+import fused_weight_gradient_mlp_cuda
+import torch.nn as nn
+import sys
+import math, pdb
+from torch.testing._internal import common_utils
+
+torch.backends.cuda.matmul.allow_tf32 = False
+
+class TestFusedWeightGradDenseCuda(common_utils.TestCase):
+    SEED: int = 123456
+    configs = [
+        #BATCH_SIZE, SEQUENCE_LENGTH, HIDDEN_SIZE, OUTPUT_SIZE
+        (1, 14, 1531, 500),
+        (1, 32, 1536, 256),
+        (1, 64, 3584, 512),
+        (1, 128, 4096, 1024),
+        (8, 32, 1536, 256),
+        (8, 64, 3584, 2048),
+        (8, 128, 4096, 4096),       
+    ]
+    def cosin_similarity(self, vector_a, vector_b):
```

Comment:
Is there a specific reason why we use this custom `cosin_similarity`? Can we just use `torch.all_close`?